engine/src/main/java/org/terasology/entitySystem/entity/EntityBuilder.java

```diff
@@ -89,7 +76,14 @@ public boolean addPrefab(String prefabName) {
     public void addPrefab(Prefab prefab) {
         if (prefab != null) {
             for (Component component : prefab.iterateComponents()) {
-                addComponent(entityManager.getComponentLibrary().copy(component));
+                Component componentCopy = entityManager.getComponentLibrary().copy(component);
+                try {
+                    addComponent(componentCopy);
+                } catch (NullPointerException e) {
+                    throw new NullPointerException(
+                            String.format("Component %s not registered (in prefab %s)", component, prefab)
+                    );
+                }
```

Comment:
I think it must have been the latter, as it would have been unusual for me to write it like this if the local variable `componentCopy` was null.

If we're not going to upgrade to JDK >= 14 to get better NPE messages, is there a pattern you prefer to catch-and-throw?

(This is also a thing I added in the course of debugging but is tangential to the rest of the PR.)

Reply:
>  is there a pattern you prefer to catch-and-throw?

hm, not right away :see_no_evil: and better error messages are always good, so not a real complaint here :wink:

Reply:
I'm reviewing this now that I'm back at the computer with the IDE and your initial hunch was right, I _did_ do the weird thing here.

I blame frusturation-and-or-sleep-deprivation-driven-development, and that's part of why we have code reviews. :wink: 

Part of my uncertainty is that so much stuff returns `null` around here that I wasn't at all clear when `null` is really an error case, but if we're getting NPEs it clearly is.

I guess this should probably be a guava `verifyNotNull` instead?

Reply:
Yes, I think `checkNotNull(componentCopy, "Component %s not registered (in prefab %s)", component, prefab)` would work here (not sure about the exact syntax...).